HST/finalize_document.py

```diff
@@ -0,0 +1,163 @@
+from hst_helper.fs_utils import (create_col_dir_in_bundle,
+                                 get_deliverable_path,
+                                 get_formatted_proposal_id,
+                                 get_program_dir_path)
+from hst_helper.general_utils import (create_xml_label,
+                                      create_collection_label,
+                                      create_csv,
+                                      get_instrument_id_set,
+                                      get_mod_history_from_label)
+
+DOC_LABEL_TEMPLATE = 'DOCUMENT_LABEL.xml'
+COL_DOC_LABEL_TEMPLATE = 'DOCUMENT_COLLECTION_LABEL.xml'
+CSV_FILENAME = 'collection.csv'
+COL_DOC_LABEL = 'collection.xml'
+
+def label_hst_document_directory(proposal_id, data_dict, logger=None, testing=False):
+    """With a given proposal id, create document directory in the final bundle. Return a
+    tupel of the path of document collection label and the path of document label. These
+    are the actions performed:
+    1. Create document directory.
+    2. Move/copy proposal files over from pipeline directory.
+    3. Create document label.
+    4. Create document collection csv.
+    5. Create document collection label.
+
+    Inputs:
+        proposal_id:    a proposal id.
+        data_dict:      a data dictionary used to create the label.
+        logger:         pdslogger to use; None for default EasyLogger.
+        testing:        the flag used to determine if we are calling the function for
+                        testing purpose with the test directory.
+    """
+    logger = logger or pdslogger.EasyLogger()
+    logger.info(f'Label hst document directory with proposal id: {proposal_id}')
+    try:
+        proposal_id = int(proposal_id)
+    except ValueError:
+        logger.exception(ValueError)
+        raise ValueError(f'Proposal id: {proposal_id} is not valid.')
+
+    formatted_proposal_id = get_formatted_proposal_id(proposal_id)
+
+    # Create document directory and move proposal files over
+    logger.info(f'Create document directory for proposal id: {proposal_id}.')
+    _, document_dir = create_col_dir_in_bundle(proposal_id, 'document', testing)
+
+    # For testing purpose, data_dict is pre-constructed, no need to walk through
+    # directories that might not exist.
+    version_id = (1, 0)
+    proposal_files_li = []
+    if not testing:
+        # Search for proposal files & program info file stored at pipeline directory
+        # Move them to bundles directory and collect neccessary data for the label
+        pipeline_dir = get_program_dir_path(proposal_id, None, root_dir='pipeline')
+        logger.info('Move over proposal files.')
+
+        for file in os.listdir(pipeline_dir):
+            basename, _, ext = file.rpartition('.')
+            if ext in DOCUMENT_EXT or file == PROGRAM_INFO_FILE:
+                proposal_files_li.append((basename, file))
+                fp = pipeline_dir + f'/{file}'
```

Comment:
Normally the variable `fp` means "file pointer" (although admittedly this is more of a C convention). For example: https://stackoverflow.com/questions/32049884/file-pointer-in-c

This code confused me for a little until I realized `fp` was a string. It might be nice to name it something else.